src/api/app/services/action_sourcediff_parser.rb

```diff
@@ -0,0 +1,74 @@
+class ActionSourcediffParser
+  attr_accessor :action
+
+  def initialize(action_sourcediff:)
+    @action_sourcediff = action_sourcediff
+  end
+
+  def call
+    @action_sourcediff&.each do |sourcediff|
+      sourcediff['filenames']&.each do |filename|
+        content = sourcediff['files'][filename].dig('diff', '_content')
+
+        sourcediff['files'][filename]['diff']['parsed_lines'] = parse_file_diffs(content)
+      end
+    end
+  end
+
+  private
+
+  def parse_file_diffs(content)
```

Comment:
```suggestion
  def parse_file_diffs(content = [])
```
With this, you can remove the ampersand in `content&`

Reply:
Done. Thanks!